internal/controller/serving/reconcilers/kserve_isvc_gateway_reconciler.go

```diff
@@ -67,7 +67,7 @@ func (r *KserveGatewayReconciler) Reconcile(ctx context.Context, log logr.Logger
 	_, meshNamespace = utils.GetIstioControlPlaneName(ctx, r.client)
 
 	// return if Address.URL is not set
-	if isvc.Status.Address != nil && isvc.Status.Address.URL == nil {
+	if isvc.Status.Address == nil || isvc.Status.Address.URL == nil {
 		log.V(1).Info("Waiting for the URL as the InferenceService is not ready yet")
```

Comment:
My only thought is - do we need to give a different error message if the Address field isn't set instead of just the URL? It seems like the original check was making sure that the Address field existed first but if the Address and URL field are always set together then this doesn't bother me too much.